src/DataTable.js

```diff
@@ -14,25 +14,58 @@ import "./DataTable.scss";
  * @prop {string} title A title for the table. Optional.
  * @prop {string} subtitle A subtitle for the table. Optional.
  * @prop {boolean} isLoading Whether the data is still loading, in which case show a spinner.
+ * @prop {(function|null)} onCheckRows If a function is provided, a checkbox will be shown for each row.
+ * On change of any checkbox elements, an array of all checked row objects will be passed to the function. By default, null.
  */
 export default function DataTable(props) {
     const {
         left, top, width, height,
         rows = [], columns = [],
         title = "Data Preview",
         subtitle,
-        isLoading = false
+        isLoading = false,
+        onCheckRows = null
     } = props;
 
-    const head = columns.map((c, j) => {
-        return <th key={j}>{c}</th>;
-    });
+    // Store the currently-checked row indices in a mutable set object.
+    const checkedRowIndicesRef = useRef(new Set());
```

Comment:
Can you give me a hint when is a good situation to use `useRef` over `useState` and member variables? I find parts of `hooks` are still unclear to me in terms of when to use A and when to use B. Does the `useRef` proper to use when we just want to store some information but do not want to update the view every time we store/change the info?

Reply:
Exactly, I think as soon as a value becomes a dependency of the view / rendering process, then `useState` is the right hook to use. In this case we do not want there to be a re-render every time a checkbox is checked, so `useRef` is better, since updating its value should not cause a re-render.